### PR TITLE
Implement Vehicle Inline Draw

### PR DESCRIPTION
**Pull request by duncanspumpkin**
These haven't been implemented quite like vanilla so will need a bit of testing.

### CODE REVIEW

File: src/OpenLoco/src/Vehicles/VehicleDraw.cpp [hunk 1]
```diff
@@ -403,18 +521,118 @@ namespace OpenLoco
...
+
+        for (auto& item : screenDistDrawItems.items)
+        {
+            drawingCtx.drawImage(loc + Ui::Point(item.dist, 0), item.image);
+        }
+        return screenDistDrawItems.totalDistance;
+    }
+
+    // 0x004B6D93
+    int16_t drawVehicleInline(Gfx::DrawingContext* drawingCtx, const Vehicles::Car& car, Ui::Point loc, uint8_t flags, Colour disableColour)
```
Comment (ZehMatt) [inline src/OpenLoco/src/Vehicles/VehicleDraw.cpp:592]:
```suggestion
    int16_t drawVehicleInline(Gfx::DrawingContext& drawingCtx, const Vehicles::Car& car, Ui::Point loc, uint8_t flags, Colour disableColour)
```

Comment (duncanspumpkin) [reply src/OpenLoco/src/Vehicles/VehicleDraw.cpp:592]: Its like that on purpose although i think i might redesign it and break out the get width version.

**Review by ZehMatt — APPROVED** [review]

**Comment by AaronVanGeffen** [comment]
Nice work! I think this is almost there, but the vehicle component offset appears to be off by one pixel. The difference will be obvious when you quickly swap between the two screenshots below.

On master:
vehicle_list_master

This PR:
vehicle_list_duncan

**Review by AaronVanGeffen — CHANGES_REQUESTED** [review]
Nice work! I think this is almost there, but the vehicle component offset appears to be off by one pixel, which adds up. The difference will be obvious when you quickly swap between the two screenshots below.

On master:
vehicle_list_master

This PR:
vehicle_list_duncan

**Comment by duncanspumpkin** [comment]
> Nice work! I think this is almost there, but the vehicle component offset appears to be off by one pixel, which adds up. The difference will be obvious when you quickly swap between the two screenshots below.
> 
> On master: vehicle_list_master
> 
> This PR: vehicle_list_duncan

vanilla used a more accurate sine table to calculate lengths. I suspect that is the source of the issue. Do we want that same increased accuracy? I was just trying to standardise things a bit more.

**Comment by LeftofZen** [comment]
> Do we want that same increased accuracy? I was just trying to standardise things a bit more.

Just going off of Aaron's screenshots, the inaccuracy seems to decrease the further down the list you go? Does this mean at the end of a *very* long list of vehicles, the drawing starts getting messed up (overlapping vehicles)? or because it's sine, it just oscillates and never goes a couple pixels off?

**Comment by ZehMatt** [comment]
> > Nice work! I think this is almost there, but the vehicle component offset appears to be off by one pixel, which adds up. The difference will be obvious when you quickly swap between the two screenshots below.
> > On master: vehicle_list_master
> > This PR: vehicle_list_duncan
> 
> vanilla used a more accurate sine table to calculate lengths. I suspect that is the source of the issue. Do we want that same increased accuracy? I was just trying to standardise things a bit more.

I think if we are going to standardise things then we should use the more accurate version, at least for rendering those, when it affects the simulation we should perhaps stick to what it currently has until we have at least everything implemented to get a new baseline.

**Comment by duncanspumpkin** [comment]
[compare.zip](https://github.com/user-attachments/files/16183210/compare.zip)
Okay i've compared the images using different sine tables. If i use the more accurate calculation the output is mostly the same as vanilla for bodies. (It looks like the rear bogie though is not in the correct position!). 

So all i need to do is add an additional more accurate sine table (and fix the rear bogie)

**Comment by duncanspumpkin** [comment]
I've looked at the rear bogie calculation and its really annoying to make it match vanilla. In vanilla it loses more precision in the calculation than the way i have it setup.
vanilla:
```
rearbogie = (0x9C * 23170) / 65536 - (0x1A * 23170) / 65536
rearbogie = 0x2d
```
myversion:
```
rearbogie = ((0x9C - 0x1A) * 23170) / 65536
rearbogie = 0x2e
```
It would be quite a pain to make this worse due to how i've split up the function so I'm proposing i do not fix the rear bogie position.

**Comment by duncanspumpkin** [comment]
So turns out i was losing a load of precision in my final calc. Just rejigging it slightly makes it match vanilla (back bogie is still not exactly the same though). So in the end i don't need more accurate sine table.

**Review by AaronVanGeffen — APPROVED** [review]
Nice work. Looks fine for me now!

On master:
vehicle_list_master

This PR:
Screenshot

**Pull request merged**